controllers/provider-azure/pkg/controller/worker/machines.go

```diff
@@ -136,6 +136,18 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
+		osDisk := map[string]interface{}{
+			"size": volumeSize,
+		}
+
+		// If no volume type is passed or the passed volume type is not a valid Azure volume type
+		// then nothing will be written into the machineclass and the default volume type
+		// of the belonging machine type will be used.
+		if pool.Volume.Type != nil {
+			if *pool.Volume.Type == "Standard_LRS" || *pool.Volume.Type == "StandardSSD_LRS" || *pool.Volume.Type == "Premium_LRS" {
```

Comment:
How about to remove this validation and let the gardener landscape operator to define the supported volume types in the `CloudProfile`? In this way the `provider-azure` extension will not need to be rebuild and released if a new disk type is added or current one is removed by Azure. Just get an example how `MachineType` is implemented.

Reply:
Yep, sound reasonable.

Reply:
+1, let's not hard-code the types here. Your documentation clearly states that only these types are valid, so using any other type is a configuration mistake and will probably prevent the machine from booting, right?

Reply:
Yes, I'm completely with you and I don't like it too. The reason for adding this check was to not damage existing clusters.

Old clusters were created with the default volume type of the machineType, because we don't passed the volume type information to MCM. This will change now and if an user has already configured worker pools with an invalid volume type then this invalid one will be now passed to MCM in the next reconconcilation and will the new machine can't be created. 

New machines and worker pools will be prevented from creation by validation of course, when they use an invalid machine type.

Reply:
OK, makes sense, thank you for the explanation. I think it makes sense to put this information as inline comment in the code itself as well.

Reply:
Done. I have extended the inline comment.

Reply:
Thanks for the details, @dkistner !